### DISCUSSION TITLE
Anyone else has problems with autolink-literal in remark-gfm?

### DISCUSSION THREAD

**Discussion by bxt** [remarkjs/remark#615] [Q&A] [answered]
I found on my machine remark-gfm is too aggressive when it comes to auto linking URL literals. E.g. `[http://www.example.com](http://www.example.com)` gets transformed into `\\[<http://www.example.com](http://www.example.com)>\n` when it should stay unchanged. So actually the link gets angle brackets `<>` even before the markdown link is recognized.

I put the following into the node console:

```
var remark = require('remark');
var gfm = require('remark-gfm');
remark().use(gfm).process('[http://www.example.com](http://www.example.com)', function(err, file) {console.log(file)})
// Prints a VFile with contents: '\\[<http://www.example.com](http://www.example.com)>\n'
```

(Skip the `.use(gfm)` and everything is fine)

Then I went ahead and cloned `remark-gfm` to produce a failing test around that, and to my surprise the test suite was already broken. It seems [the test case with `https://example`](https://github.com/remarkjs/remark-gfm/blame/main/test/fixtures/autolink-literal/input.md#L55) gets converted to `<https://example>` instead of `https\://example`. I even tried `node@12.20.1` and `node@14.15.4`.

I saw on https://astexplorer.net/ everything works fine.

So I was wondering if something is wrong with my machine.

And then I saw that the master build pipeline of remark-gfm ran last 3 months ago https://travis-ci.org/github/remarkjs/remark-gfm/builds/735687923 and it has no `package-lock.json` or `yarn.lock`, so it could also be a transitive dependency that introduced a different bahavior.

So what do you think? Did someone else also find strange things around the autolink-literal in remark-gfm? Maybe if some remark people are reading here, they could just re-start the build pipeline.

**Comment by ChristianMurphy** [comment, marked as answer]
This is being tracked in https://github.com/micromark/micromark-extension-gfm-autolink-literal/issues/5 and https://github.com/micromark/micromark-extension-gfm-autolink-literal/pull/6

**Reply by bxt** [reply]
Wow that was fast, thanks! Indeed that seems to be the issue, I just looked in the wrong repos 🙈 sorry!

**Reply by wooorm** [reply]
no problem @bxt! For some weird reason, three people walked into this issue just today. So thanks for reporting, definitely something to fix!
